### DISCUSSION TITLE
Source generator Newtonsoft.Json FileNotFoundException

### DISCUSSION THREAD

**Discussion by onetsmr** [dotnet/roslyn#51090] [Q&A] [answered]
How can I use Newtonsoft.Json lib in the source generator project?

My generator project
```
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>netstandard2.0</TargetFramework>
    <LangVersion>preview</LangVersion>
    <UserSecretsId>...</UserSecretsId>
    <GeneratePackageOnBuild>true</GeneratePackageOnBuild>
    <IncludeBuildOutput>false</IncludeBuildOutput>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.CodeAnalysis.CSharp" Version="3.8.0" PrivateAssets="all" />
    <PackageReference Include="Microsoft.CodeAnalysis.Analyzers" Version="3.3.2" PrivateAssets="all" />
    <PackageReference Include="Newtonsoft.Json" Version="12.0.3" />
  </ItemGroup>

  ... missing ...

</Project>
```

I get
```
Exception was of type 'FileNotFoundException' with message 
'Could not load file or assembly 'Newtonsoft.Json, Version=12.0.0.0, 
Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed' 
or one of its dependencies.
```

What I already tried:
1. GeneratePackageOnBuild true/false
2. IncludeBuildOutput true/false
3. Add Newtonsoft.Json 12.0.3 into the main project, which use generator project (it is .net 5.0)
4. Make as described here https://github.com/dotnet/roslyn/blob/master/docs/features/source-generators.cookbook.md#use-functionality-from-nuget-packages

Nothing works. I keep getting 'Could not load file or assembly' exception.

If I comment code line which calls Newtonsoft.Json.JsonConvert class and go to the debugger - then I see Newtonsoft.Json lib in the context.Compilation.ReferencedAssemblyNames array.

**Comment by NealCallaghan** [comment, marked as answer]
Checkout the csproj here https://github.com/dotnet/roslyn-sdk/blob/master/samples/CSharp/SourceGenerators/SourceGeneratorSamples/SourceGeneratorSamples.csproj also to get to debug, in your execute method of the generator write `Debugger.Launch()` then you can debug. VS will ask you which instance etc but you will get it.
